### ISSUE TITLE
[LocalizeManager] Overriding certain keys in an already existing namespace

### ISSUE THREAD

**Issue by ElenaMirunaGagea** [ing-bank/lion#1790]
**Our Need**

We are developing a feature that allows our users to override certain existing translations from our web components depending on what business requirements they may have.

Our expected behaviour is that our web component will provide a default set of translations and the app using the component can override them by replacing certain keys in the existing namespace (of the component).

Current usage and limitation:

```
  static get localizeNamespaces() {
    // using an explicit loader function
    return [
      { 'ep-means-login': locale => import(`./translation/${locale}.js`) },
      ...super.localizeNamespaces,
    ];
  }
```

In the current implementation of our components we are using the [LocalizeMixin.localizeNamespaces()](https://github.com/ing-bank/lion/blob/8890cc0dea16c168485c1edebb1164a8f5ac2b55/packages/localize/src/LocalizeMixin.js#L20) method to register our namespaces containing the translations. This method is calling, on a lower level, the [LocalizeManager.addData()](https://github.com/ing-bank/lion/blob/8890cc0dea16c168485c1edebb1164a8f5ac2b55/packages/localize/src/LocalizeManager.js#L204) method that checks to see if the namespace already exists before adding it. Because of this existing check any namespace overrides are prevented.

**Proposed solution**

To be able to override certain keys in an already existing namespace we propose to add a new method to the LocalizeManager:

```
/**
 * @param {string} locale
 * @param {string} namespace
 * @param {object} data
 */
updateData(locale, namespace, data) {
  this.__storage[locale] = this.__storage[locale] || {};
  this.__storage[locale][namespace] = {
	…this.__storage[locale][namespace],
	…data
   };
}
```

This method will not be a breaking change and it will keep the backwards compatibility.
To override the namespace the user will need to import the LocalizeManager and call the updateData method:

```
LocalizeManager.updateData(“en-GB”, “ep-means-login”, {
	MyKey: “MyValue”
});
```

I’m looking forward for your feedback.
Thank you!

**Comment by DannyMoerkerke** [comment]
What exactly is your use case? Why would the application have to override certain keys?

**Comment by DannyMoerkerke** [comment]
This has been fixed in 0.1.1.

`LocalizeManager` can now be instantiated with the `allowOverridesForExistingNamespaces` option which defaults to `false`.
When it's set to `true`, existing data in a namespace for a give locale can be overridden in the `addData` method.
